src/components/routerView.vue

```diff
@@ -1,94 +1,69 @@
 <template>
   <template v-if="component">
-    <slot name="default" v-bind="{ route, component, rejection }">
-      <component :is="component" />
+    <slot name="default" v-bind="{ route, component, props, rejection }">
+      <template v-if="rejection">
+        <component :is="rejection.component" />
+      </template>
+      <template v-else>
+        <component :is="component" :props="props" />
+      </template>
     </slot>
   </template>
 </template>
 
-<script lang="ts">
-/**
- * @ignore
- */
-</script>
-
 <script lang="ts" setup>
-  import { Component, UnwrapRef, VNode, computed, provide, resolveComponent } from 'vue'
+  import { Component, UnwrapRef, VNode, computed, provide } from 'vue'
   import { usePropStore } from '@/compositions/usePropStore'
   import { useRejection } from '@/compositions/useRejection'
   import { useRoute } from '@/compositions/useRoute'
   import { useRouterDepth } from '@/compositions/useRouterDepth'
-  import { component as componentUtil } from '@/services/component'
   import { RouterRejection } from '@/services/createRouterReject'
   import { RouterRoute } from '@/services/createRouterRoute'
-  import { CreateRouteOptions, isWithComponent, isWithComponents } from '@/types/createRouteOptions'
   import { depthInjectionKey } from '@/types/injectionDepth'
+  import { useComponentsStore } from '@/compositions/useComponentsStore'
 
   const { name = 'default' } = defineProps<{
     name?: string,
+    props?: unknown,
```

Comment:
Not sure about this. When a `RouterView` was rendered automatically it still ends up trying to pass props too it. But since this component doesn't have a prop called props vue would put a warning in the console "extraneous non props warning". Adding this hear silences that warning. Don't like it though. Will have to figure out a better way to silence the warning.

Reply:
Resolved by https://github.com/kitbagjs/router/pull/440/commits/8f2d7208d984f32c872863ddad2ad90f346a475f